### PR TITLE
Feat/invalid email mapping

### PR DESCRIPTION
**Pull request by apwidejulien**
We have records where use provided an invalid email address format in MPAC. (eg: firstname.lastname@domain.coom)

This PR propose a mapping capability to substitute mpac invalid email address with a fixed one which can be correctly imported into hubspot:
- firstname.lastname@domain.coom => firstname.lastname@domain.com

### CODE REVIEW

**Comment by boris-moduscreate** [comment]
@nhan-nguyen-se Can you please review?

File: src/lib/config/env.ts [hunk 1]
```diff
@@ -108,6 +108,11 @@ export function hubspotContactConfigFromENV(): HubspotContactConfig {
 export function mpacConfigFromENV(): MpacConfig {
   return {
     ignoredEmails: new Set((optional('IGNORED_EMAILS')?.split(',') ?? []).map((e) => e.toLowerCase())),
+    emailMappings: optional('EMAIL_MAPPINGS')?.split(',').reduce((p: any, c: string) => {
```
Comment (nhan-nguyen-se) [inline src/lib/config/env.ts:111]: I guess `p` is of type `Record<string, string>` right? We should avoid `any` if possible.

File: src/lib/util/helpers.ts [hunk 2]
```diff
@@ -4,7 +4,7 @@ export function isPresent<T>(o: T | undefined | null): o is T {
 
 export function batchesOf<T>(array: T[], count: number): T[][] {
   const tmp = [...array];
-  const batches = [];
+  const batches: Array<any> = [];
```
Comment (nhan-nguyen-se) [inline src/lib/util/helpers.ts:7]: Can we use `T[]` as a type for `batches`?

**Review by nhan-nguyen-se — CHANGES_REQUESTED** [review]

**Comment by nhan-nguyen-se** [comment]
Just 2 minors typescript change request. The logic looks good to me.

File: src/lib/config/env.ts (hunk 1 above)
Comment (apwidejulien) [reply src/lib/config/env.ts]: done

File: src/lib/util/helpers.ts (hunk 2 above)
Comment (apwidejulien) [reply src/lib/util/helpers.ts]: done

**Review by nhan-nguyen-se — APPROVED** [review]

**Comment by boris-moduscreate** [comment]
@apwidejulien after releasing this and configuring 	`HUBSPOT_DEAL_TRANSACTIONLINEITEMID_ATTR` I still get an error saying it's not configured. Any thoughts on what may be happening here?

**Comment by boris-moduscreate** [comment]
Hmm maybe something weird in my environment, but it works when I place it in the same order as the definition. If you can't reproduce then lets not worry about it.

**Pull request merged**